multimodal/src/autogluon/multimodal/utils/onnx.py

```diff
@@ -7,99 +7,70 @@
 
 logger = logging.getLogger(__name__)
 
+try:
+    import tensorrt  # Unused but required by TensorrtExecutionProvider
+except:
+    logger.warning(
+        "Failed to import tensorrt package. "
+        "onnxruntime would fallback to CUDAExecutionProvider instead of using TensorrtExecutionProvider."
+    )
 
-def get_onnx_input(pipeline: str, config: Optional[Dict] = None):
-    """
-    Get information for a predictor to export its model in onnx format.
 
-    Parameters
-    ----------
-    pipeline
-        The predictor's pipeline.
-    config
-        The predictor's config.
+def onnx_get_dynamic_axes(input_keys: List[str]):
+    dynamic_axes = {}
+    for k in input_keys:
+        if "token_ids" in k or "segment_ids" in k:
+            dynamic_axes[k] = {0: "batch_size", 1: "seq_length"}
+        elif "valid_length" in k or k.startswith("numerical") or k.startswith("timm_image"):
+            dynamic_axes[k] = {0: "batch_size"}
```

Comment:
The shape of `images` in `timm_image`'s input should be `(b, n, c, h, w)`, where both b and n may be dynamic.

Reply:
I would suggest avoid adding image size into dynamic dimensions, because
1. adding an extra dynamic dimension would increase the complexity of model compilation
2. the compiled model with too many dynamic dimensions could be suboptimal in terms of performance

The question becomes do we really need to support dynamic shape in image data? If yes, what is the lower bound and upper bound of the image size.

Reply:
In a pretrained setting, n should be a fixed number, since input DataFrame should have same number of image columns, isn't it?